Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+   - Limits
+ * Ceph components and daemons, basic Ceph configuration
+ * Rook basics and how to install Rook-Ceph
+
+In places, we will give examples that describe an imaginary data center. This data center is
+hypothetical, and it will focus on the Ceph- and Rook-centric elements and ignore user applications.
+
+Our example data center has two rooms for data storage. A properly fault tolerant Ceph cluster
+should have at least three monitor (MON) nodes. These nodes should be spread across fault-tolerant
+rooms if possible. Real world data centers commonly have two rooms, so the example will have a
+separate failure domain just for the third monitor node. As such, our hypothetical data center has
+two rooms and one failure domain, with the following configuration:
+* The failure domain is small and can only to be used for the third Ceph MON;
+  it does not have space for storage nodes.
+* Eight Ceph OSD nodes provide a good amount of data safety without requiring too many nodes.
+* These eight nodes should be equally separated — four to each data center room.
+* We can imagine that the four nodes are separated in each room into two racks.
+* In the event of a MON node failure, let's decide we want to be able to run MONs on each rack
+
+The data center looks as follows:
+
+![Example Datacenter](media/ceph-bp-example-datacenter.png)
+
+Now we will dig a little deeper and talk about the actual disks used for Rook and Ceph storage. To
+ensure we are following known Ceph best practices for this data center setup, ensure that MON
+storage goes on the SSDs. We'll discuss this a little more later. Because each rack should be able
+to run a Ceph MON, one of the nodes in each rack will have an SSD that is usable for MON storage.
+Additionally, all nodes in all racks (except in the failure domain) will have disks for OSD storage.
+This will look like the following:
+
+![Example Datacenter - Disks](media/ceph-bp-example-datacenter-disks.png)
+
+> **IMPORTANT:** Refer to these diagrams when we discuss the example data center below.
+
+
+## Introduction
+
+Ceph and Kubernetes both have their own well-known and established best practices. Rook bridges the
+gap between Ceph and Kubernetes, putting it in a unique domain with its own best practices to
+follow. Therefore, this document specifically covers best practices for running Ceph on Kubernetes
+with Rook. Because Rook augments on top of Kubernetes, it has different ways of meeting both Ceph
+and Kubernetes best practices. This is in comparison to the bare metal versions of each. Out of the
+box, Rook is predominantly a default Ceph cluster. The Ceph cluster needs tuning to meet user
+workloads, and Rook does not absolve the user from planning out their production storage cluster
+beforehand.
+
+For the purpose of this document, we will consider two simplified use cases to help us make informed
+decisions about Rook and Ceph:
+ * Co-located: User applications co-exist on nodes running Ceph
```

Comment:
I had a lot of back and forth with our product manager. Hyperconverged also suggests a unified control plane I'm told, and since this doesn't have that, it is technically "co-located". I had this as "hyperconverged" in my first drafts, and I personally don't know if I see evidence in regular google searches for that distinction. Personally, I'm fine with this being "hyperconverged", but it makes my life a little easier not to make the change. If you're on the fence, I'd rather keep it as-is, but if you have a preference still, I'll go with that.

Reply:
That's true, I've seen cases where Hyperconverged included the control plane too. I guess I'm fine with co-located too :) no need to change it.